echo-core/src/main/groovy/com/netflix/spinnaker/echo/artifacts/MessageArtifactTranslator.java

```diff
@@ -87,7 +88,11 @@ private Map readMapValue(ObjectMapper mapper, String messagePayload) {
     try {
       return mapper.readValue(hydratedTemplate, artifactListReference);
     } catch (IOException ioe) {
-      throw new RuntimeException(ioe);
+      // Failure to parse artifacts from the message indicates either
+      // the message payload does not match the provided template or
+      // there is no template and no artifacts are expected
+      log.error("Unable to parse artifact from {}", hydratedTemplate);
```

Comment:
Allows messages with no artifacts in them to pass through for all subscription types.